Makefile

```diff
@@ -1,42 +1,46 @@
-.DEFAULT-GOAL := help
+CRYSTAL=/usr/bin/crystal
+CRYSTAL_FLAGS=--release
 
-# Variables
-PROJECT_NAME?=$$(git remote -v | head -n1 | awk '{print $$2}' | sed 's/.*\///' | sed 's/\.git//')
-CWD=$(shell pwd)
-VERSION_NUMBER?=$$(docker run -e TYPE=semver -v $(CWD):/repo codacy/ci-git-version:latest)## lazy. run with VERSION_NUMBER=x to prevent slow runs
+all: clean fmt test build docker_build ## clean and produce target binary and docker image
 
-.PHONY: help
-help:
-	@echo "---------------------------------------------------------------------------------------------------------"
-	@echo "build and deploy help"
-	@echo "---------------------------------------------------------------------------------------------------------"
-	@grep -E '^[a-zA-Z_/%\-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
-	@echo "---------------------------------------------------------------------------------------------------------"
+test: ## runs crystal tests
+	$(CRYSTAL) spec spec/*.cr
+
+fmt: ## format the crystal sources
+	$(CRYSTAL) tool format
+
+build: ## compiles from crystal sources
+	mkdir -p bin
```

Comment:
you could prepend `-` to mkdir so that the build doesn't fail if `bin/` already exits

Reply:
`-p` does the same job